### ISSUE TITLE
Error on Python 3 for password:  sequence item 1: expected bytes, bytearray, or an object with the buffer interface, str found

### ISSUE THREAD

**Issue by xuhdev** [OfflineIMAP/offlineimap#389]
#### General informations
- system/distribution (with version):  Debian Jessie
- offlineimap version (`offlineimap -V`): offlineimap v7.0.8, imaplib2 v2.55 (bundled), Python v3.4.2
- Python version: 3.4.2
#### Configuration file offlineimaprc

```
[general]
accounts = test

[Account test]
# autorefresh = 2
localrepository = testLocal
remoterepository = testRemote
synclabels = yes
quick = 5

[Repository testLocal]
type = Maildir
localfolders = Maildir

[Repository testRemote]
type = IMAP
remotehost = imap.zoho.com
remoteport = 993
ssl = yes
remoteuser = blabla
maxconnections = 2
idlefolders = ['INBOX']
sslcacertfile = /etc/ssl/certs/ca-certificates.crt
remotepass = blabla
```
#### Logs, error

```
Account sync test:
 *** Processing account test
 Establishing connection to imap.zoho.com:993 (testRemote)
 ERROR: While attempting to sync account 'test'
  sequence item 1: expected bytes, bytearray, or an object with the buffer interface, str found
 *** Finished account 'test' in 0:01
ERROR: Exceptions occurred during the run!
ERROR: While attempting to sync account 'test'
  sequence item 1: expected bytes, bytearray, or an object with the buffer interface, str found

Traceback:
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/accounts.py", line 271, in syncrunner
    self.__sync()
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/accounts.py", line 334, in __sync
    remoterepos.getfolders()
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/repository/IMAP.py", line 452, in getfolders
    imapobj = self.imapserver.acquireconnection()
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/imapserver.py", line 540, in acquireconnection
    self.__authn_helper(imapobj)
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/imapserver.py", line 406, in __authn_helper
    if func(imapobj):
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/imapserver.py", line 336, in __authn_plain
    imapobj.authenticate('PLAIN', self.__plainhandler)
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/bundled_imaplib2.py", line 705, in authenticate
    typ, dat = self._simple_command('AUTHENTICATE', mechanism.upper())
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/bundled_imaplib2.py", line 1692, in _simple_command
    return self._command_complete(self._command(name, *args), kw)
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/bundled_imaplib2.py", line 1418, in _command
    literal = literator(data, rqb)
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/bundled_imaplib2.py", line 2283, in process
    ret = self.mech(self.decode(data))
  File "/home/hong/.venv/offlineimap/lib/python3.4/site-packages/offlineimap/imapserver.py", line 202, in __plainhandler
    retval = NULL.join((authz, authc, passwd))
```
#### Steps to reproduce the error

Use the configuration file shown, then run offlineimap.

**Comment by zhangjunphy** [comment]
Python 3 configParser returns option values as utf-8 strings, so I guess this no longer works: https://github.com/OfflineIMAP/offlineimap/blob/d5edf2dc11d22ebfa4739dd4dd8d6edf67ff6ef2/offlineimap/repository/IMAP.py#L189-L194

**Comment by nicolas33** [comment]
You're right. Supporting py3 implies correct support of UTF-8. This is far from easy to get it right with the codebase. Also, this requires both invasive changes and changes on the critical imaplib2 library. Because of this the py3 support has somewhat stalled.

Historically, I've tried twice to correctly support UTF-8 but I stopped my work once I realized the best would be to restart from scratch. Then, @dolohow came with another strategy: support Py3 first and see how far we can go. AFAICT, he stopped too.

However, I'd welcome contributions and patches.
